### DISCUSSION TITLE
Resource Dependency requirements

### DISCUSSION THREAD

**Discussion by devlife** [PrateekKumarSingh/AzViz#36] [General]
I did `Export-AzViz` and used all of the default values but the only resource dependency arrow drawn was from a web app to the resource group. Under what circumstances would a dependency be drawn between a web app and application insights, for example?

**Comment by PrateekKumarSingh** [comment]
Hey @devlife, right now resource dependency is drawn on basis of:

1. `DependsOn` property in the Resource Group ARM template which is exported/parsed automatically to obtain this property
2. If a Network Watcher is detected in the Location of the target Resource Group, then it can run `Get-AzNetworkWatcherTopology` cmdlet to obtain network associations/dependency like `VM -> NIC`

Then both of them are overlaid to show edge representation, marked differently also shown in the diagram legend.

I hope this answers your question, and I'll add more documentation to make this point clear.

**Comment by NickNiebling** [comment]
@PrateekKumarSingh does e.g. VM and NIC have to be in same resource group?
